internal/querycoordv2/observers/leader_observer.go

```diff
@@ -57,28 +58,10 @@ func (o *LeaderObserver) Start() {
 	ctx, cancel := context.WithCancel(context.Background())
 	o.cancel = cancel
 
+	o.dispatcher.Start()
+
 	o.wg.Add(1)
-	go func() {
-		defer o.wg.Done()
-		ticker := time.NewTicker(interval)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ctx.Done():
-				log.Info("stop leader observer")
-				return
-
-			case req := <-o.manualCheck:
-				log.Info("triggering manual check")
-				ret := o.observeCollection(ctx, req.CollectionID)
-				req.Notifier <- ret
-				log.Info("manual check done", zap.Bool("result", ret))
-
-			case <-ticker.C:
-				o.observe(ctx)
-			}
-		}
-	}()
+	go o.schedule(ctx)
```

Comment:
The caller should maintain the waitgroup by itself, not the callee
```golang
go func() {
defer o.wg.Done()
o.schedule(ctx)
}()
```

Reply:
nope, `wg` shall be `Add`ed and `Wait`ed by the owner, which is the observer, otherwize there will be race on it

Reply:
fixed